proposal/custom-url-resolution.md

```diff
@@ -0,0 +1,128 @@
+# Custom url resolution: Draft 1
+
+_[(Issue)](https://github.com/sass/sass/issues/2535)_
+
+## Table of Contents
+
+- [Background](#background)
+- [Summary](#summary)
+  - [Example](#example)
+- [Semantics](#semantics)
+  - [Steps](#steps)
+  - [JavaScript API](#javascript-api)
+  - [Using Variables](#using-variables)
+
+## Background
+
+> This section is non-normative.
+
+Many css features require the use of a url reference to reference resources from outside the sass files, however these files also need to exist on the eventual output directory and server. To ensure the references are valid, the sass API should allow for the user to provide a way to remap and/or inline these resources.
+
+## Summary
+
+> This section is non-normative.
+
+This proposal defines a standardized way to remap the url references to the final location on the server or inline reference.
+
+This is accomplished by running url references through the url rewriting plugin if a url rewriting plugin has been defined.
+
+### Example
+
+This example demonstrates the possibility of rewriting urls to inline assets using base64 and data URIs.
+
+Input:
+
+```scss
+.upload-icon {
+  background-url: url(../icons/upload-icon.svg);
+}
+```
+
+Output:
+
+```scss
+.upload-icon {
+  background-url: url(data:image/svg+xml;base64,Jggg==);
+}
+```
+
+## Semantics
+
+This proposal defines new behavior for `url()`, to allow users to rewrite url references to how it will be accessible on the browser.
```

Comment:
I do strongly feel that we should define a new `sass-url()` function rather than overriding the existing `url()` function. It's also much simpler to specify the behavior of a function than a syntactic construct.

Reply:
alright probably a good idea, changed it